File: frontend/src/validations/courseSchema.ts
import { z } from "zod";

export const courseSchema = z.object({
    title: z.string().min(3, "Title is required"),

    slug: z.string().min(3, "Slug is required"),

    short_description: z
        .string()
        .min(5, "Short Description is required"),

    description: z
        .string()
        .min(10, "Description is required"),

    category: z.string().min(2),

    level: z.string().min(2),

    language: z.string().min(2),

    duration: z.string().min(2),

    instructor: z.string().min(2),

    price: z.coerce
        .number()
        .min(0, "Price must be greater than 0"),

    thumbnail: z.string().optional(),

    is_published: z.boolean(),
});

export type CourseFormData = z.infer<typeof courseSchema>;

File: backend/schemas/course_schema.py
from pydantic import BaseModel, ConfigDict
from decimal import Decimal
from uuid import UUID
from typing import Optional


class CourseCreate(BaseModel):
    title: str
    slug: str
    short_description: str
    description: str
    thumbnail: Optional[str] = None
    category: str
    level: str
    language: str = "English"
    duration: str
    price: Decimal = Decimal("0.00")
    instructor: str
    is_published: bool = False


class CourseUpdate(BaseModel):
    title: Optional[str] = None
    slug: Optional[str] = None
    short_description: Optional[str] = None
    description: Optional[str] = None
    thumbnail: Optional[str] = None
    category: Optional[str] = None
    level: Optional[str] = None
    language: Optional[str] = None
    duration: Optional[str] = None
    price: Optional[Decimal] = None
    instructor: Optional[str] = None
    is_published: Optional[bool] = None


class CourseResponse(BaseModel):
    id: UUID
    title: str
    slug: str
    short_description: str
    description: str
    thumbnail: Optional[str]
    category: str
    level: str
    language: str
    duration: str
    price: Decimal
    instructor: str
    is_published: bool

    model_config = ConfigDict(from_attributes=True)